### PR TITLE
[DEV-6215] Fix missing release_settings in filter_jira_stuff

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
This change is [Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3473)

### CODE REVIEW

File: server/athenian/api/controllers/jira_controller/common.py [hunk 1]
```diff
@@ -79,20 +75,17 @@ async def collect_account_info(
...
-        branches = release_settings = logical_settings = None
-        default_branches = {}
+
+    settings = Settings.from_request(request, account, prefixer)
+    (branches, default_branches), logical_settings = await gather(
+        BranchMiner.load_branches(
+            repos, prefixer, account, meta_ids, mdb, pdb, cache, strip=True,
+        ),
+        settings.list_logical_repositories(repos),
+        op="sdb/branches and releases",
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/jira_controller/common.py:85]:
```suggestion
        op="sdb/branches, logical settings",
```

File: server/athenian/api/controllers/jira_controller/common.py [hunk 2]
```diff
@@ -79,20 +75,17 @@ async def collect_account_info(
...
+    )
+    repos = logical_settings.append_logical_prs(repos)
+    release_settings = await settings.list_release_matches(repos)
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/jira_controller/common.py:88]: We should call `settings.list_release_matches()` inside `gather()`, we don´t care about which repos to list

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: server/athenian/api/controllers/jira_controller/common.py (hunk 2 above)
Comment (gaetano-guerriero) [reply server/athenian/api/controllers/jira_controller/common.py:88]: should we also call `settings.list_logical_repositories` with no repos ?

File: server/athenian/api/controllers/jira_controller/common.py [hunk 3]
```diff
@@ -79,20 +75,16 @@ async def collect_account_info(
...
-        branches = release_settings = logical_settings = None
-        default_branches = {}
+
+    settings = Settings.from_request(request, account, prefixer)
+    (branches, default_branches), logical_settings, release_settings = await gather(
+        BranchMiner.load_branches(
+            repos, prefixer, account, meta_ids, mdb, pdb, cache, strip=True,
+        ),
+        settings.list_logical_repositories(repos),
+        settings.list_release_matches(),
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/jira_controller/common.py:85]: @gaetano-guerriero I think that `settings.list_release_matches(repos)` will automatically include the logical repos. We can change the code, it seems

**Pull request merged**